homeassistant/components/slide/__init__.py

```diff
@@ -0,0 +1,159 @@
+"""Component for the Go Slide API."""
+
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+
+from homeassistant.const import (
+    CONF_USERNAME,
+    CONF_PASSWORD,
+    CONF_SCAN_INTERVAL,
+    CONF_TIMEOUT,
+)
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.discovery import async_load_platform
+from homeassistant.helpers.event import async_track_time_interval, async_call_later
+from .const import (
+    DOMAIN,
+    SLIDES,
+    API,
+    COMPONENT,
+    DEFAULT_TIMEOUT,
+    CONF_RETRY,
+    DEFAULT_RETRY,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_SCAN_INTERVAL = timedelta(seconds=30)
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {
+                vol.Required(CONF_USERNAME): cv.string,
+                vol.Required(CONF_PASSWORD): cv.string,
+                vol.Optional(
+                    CONF_SCAN_INTERVAL, default=DEFAULT_SCAN_INTERVAL
+                ): cv.time_period,
+                vol.Optional(CONF_TIMEOUT, default=DEFAULT_TIMEOUT): cv.positive_int,
+                vol.Optional(CONF_RETRY, default=DEFAULT_RETRY): cv.positive_int,
+            }
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+async def async_setup(hass, config):
+    """Set up the Slide platform."""
+
+    # pylint: disable=unused-argument
+    async def update_slides(now=None):
+        """Update slide information."""
+        result = await hass.data[DOMAIN][API].slidesoverview()
+
+        if result is None:
+            _LOGGER.error("Slide API does not work or returned an error")
+            return
+
+        if result:
+            _LOGGER.debug("Slide API returned %d slide(s)", len(result))
+        else:
+            _LOGGER.warning("Slide API returned 0 slides")
+
+        for key in hass.data[DOMAIN][SLIDES]:
+            hass.data[DOMAIN][SLIDES][key]["pos"] = None
+
+        for slide in result:
+            if "device_id" not in slide:
+                _LOGGER.error(
+                    "Found invalid Slide entry, 'device_id' is " "missing. Entry=%s",
+                    slide,
+                )
+                continue
+
+            uid = slide["device_id"].replace("slide_", "")
+            slidenew = {}
+            slidenew["mac"] = uid
+            slidenew["id"] = slide["id"]
+            slidenew["name"] = slide["device_name"]
+
+            if "device_info" not in slide:
+                _LOGGER.error(
+                    "Slide %s (%s) has no 'device_info' Entry=%s",
+                    slide["id"],
+                    slidenew["mac"],
+                    slide,
+                )
+                continue
+
+            # Check if we have 'pos' (OK) or 'code' (NOK)
+            if "pos" in slide["device_info"]:
+                slidenew["online"] = True
```

Comment:
We never use the online key.

Reply:
I added the property available, so it is used now.